From 553ba13ed86f192a575bd028c74d5687a6b1db62 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Fri, 24 Oct 2014 12:28:09 -0400
Subject: [fs] ext4: revert Disable punch hole on non-extent mapped files

Message-id: <1414153689-13257-1-git-send-email-lczerner@redhat.com>
Patchwork-id: 98430
O-Subject: [RHEL 7.1 PATCH] ext4: revert Disable punch hole on non-extent mapped files
Bugzilla: 1150178
RH-Acked-by: Carlos Maiolino <cmaiolino@redhat.com>

BZ 1150178
https://bugzilla.redhat.com/show_bug.cgi?id=1150178

BREW 8157985
http://brewweb.devel.redhat.com/brew/taskinfo?taskID=8157985

This commit does not have its upstream counterpart.

This commit effectively reverts rhel commit
159d8a6cd68e4ffe44eebb9032378e1648a39bc8, because the problem with
indirectly mapped file has been fixed with commit "ext4: fix punch hole
on files with indirect mapping"

Signed-off-by: Lukas Czerner <lczerner@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/extents.c b/fs/ext4/extents.c
index 8ad0b8d..9b9e249 100644
--- a/fs/ext4/extents.c
+++ b/fs/ext4/extents.c
@@ -4908,13 +4908,6 @@ long ext4_fallocate(struct file *file, int mode, loff_t offset, loff_t len)
  ext4_lblk_t lblk;
  unsigned int blkbits = inode->i_blkbits;
 
- /*
-  * currently supporting (pre)allocate mode for extent-based
-  * files _only_
-  */
- if (!(ext4_test_inode_flag(inode, EXT4_INODE_EXTENTS)))
-  return -EOPNOTSUPP;
-
  /* Return error if mode is not supported */
  if (mode & ~(FALLOC_FL_KEEP_SIZE | FALLOC_FL_PUNCH_HOLE |
        FALLOC_FL_COLLAPSE_RANGE | FALLOC_FL_ZERO_RANGE))
@@ -4927,6 +4920,13 @@ long ext4_fallocate(struct file *file, int mode, loff_t offset, loff_t len)
  if (ret)
   return ret;
 
+ /*
+  * currently supporting (pre)allocate mode for extent-based
+  * files _only_
+  */
+ if (!(ext4_test_inode_flag(inode, EXT4_INODE_EXTENTS)))
+  return -EOPNOTSUPP;
+
  if (mode & FALLOC_FL_COLLAPSE_RANGE)
   return ext4_collapse_range(inode, offset, len);
 
-- 
1.7.1